ludwig/modules/metric_modules.py

```diff
@@ -123,11 +123,26 @@ def get_inputs(cls):
 
 
 @register_metric(ROC_AUC, [BINARY])
-class ROCAUCMetric(AUROC, LudwigMetric):
-    def __init__(self, **kwargs):
+class ROCAUCMetric(LudwigMetric):
+    """Fast implementation of metric for area under ROC curve."""
+
+    def __init__(
+        self,
+        num_thresholds: int = 201,
+        epsilon: float = 1e-7,
+        **kwargs,
+    ) -> None:
         super().__init__(dist_sync_fn=gather_all_tensors)
+        self.num_thresholds = num_thresholds
+        self.epsilon = epsilon
+        # Add an extra dimension to each state to make it compatible with Horovod all gather.
```

Comment:
What do you think about moving this fix to `gather_all_tensors`? It seems the correct behavior for all metrics should be something like:

```
t = t.reshape([1, *t.shape])
hvd.allgather(t)
```

Reply:
That makes sense -- Ip updated the code to reflect this.

I guess we weren't seeing issues from this previously as each metric being gathered was a scalar value, and so concatenating and reducing them wasn't causing an issue.

I haven't tested out this change yet but I can update here to make sure that it doesn't cause an issue with other metrics.